src/types.ts

```diff
@@ -52,7 +52,7 @@ export type InputAttributes = Omit<
 type NumberFormatProps<Props, BaseType = InputAttributes> = Props &
   Omit<InputAttributes, keyof BaseType> &
   Omit<BaseType, keyof Props | 'ref'> & {
-    customInput?: React.ComponentType<BaseType>;
+    customInput?: React.ComponentType<React.InputHTMLAttributes<HTMLInputElement> & React.RefAttributes<HTMLInputElement>>;
```

Comment:
This will force the customInput to be type of Input element. While custom Input can be any component, and the NumberFormatProps inherits from there.

Reply:
let me try to fix it, may be I need to rewrite a couple of types